design/common/multi-net-multus.md

```diff
@@ -0,0 +1,110 @@
+selector. Using host network provider, pod can use host's network namespace
+and its interfaces to use as cluster network. Using popular multi-plugin CNI
+plugin such as [multus][multus-cni], Rook operators can attach multiple network
+interfaces to pods.
+
+One thing to remember, leaving the network configuration empty will default to
+kubernetes cluster default networking.
+
+### Keys
+
+The network interface selectors with key values `public` and
+`cluster` are specified for Ceph.
+
+### Network Interface Selector
+
+Network interface selector is fed to network provider to connect pod to cluster
+network. This selector may vary from one network provider to another. For
+example, host network provider only needs to know the interface name.
+
+On the other hand, multi-plugin CNI plugin needs to know the network attachment
+definition's name and vice versa. Multi-plugin such as multus may seem to follow
+_Network Attachment Selection Annotation_ documented by Kubernetes Network Plumbing Working Group's
+[Multi-Net Spec].
+
+Any future multi-homed network provider that implements the Network Plumbing WG's Multi-net spec may
+use the `multus` network. Previously-identified providers [CNI-Genie] and [Knitter] have since gone
+dormant (upd. 2023), and Multus continues to be the most prominent known provider.
+
+## Multi-Net validation tester
+
+Rook will implement a user-runnable routine that validates a multi-net configuration based on the
+[Multi-Net Spec]. Users can input one or both of the Network Attachment Definitions (NADs) for
+Ceph's `multus` network, and the routine will perform cursory validation of the ability of the
+specified networks to support Ceph.
+
+The routine will start up **a single** web server with the specified networks attached (via NADs).
+The routine will also start up a number of clients on each node that will test the network(s)'s
+connections by HTTP(S) requests to the web server.
+
+A client represents a simplified view of a Ceph daemon from the networking perspective. The number
+of clients is intended to represent the number of Ceph daemons that could run on a node in the worst
+possible failure case where all daemons are rescheduled to a single node. This helps verify that the
+network provider is able to successfully assign IP addresses to Ceph daemon pods in the worst case.
+Some network interface software/hardware may limit the number of addresses that can be assigned to
+an interface on a given node, and this helps verify that issue is not present.
+
+It is important that the web server be only **a single** instance. If clients collocated on the same
+node with the web server can make successful HTTP(S) requests but clients on other nodes are not,
+that is an important indicator of inter-node traffic being blocked.
+
+The validation test routine will run in the Rook operator container. Rook's kubectl [Krew plugin]
+may facilitate running the test, but it is useful to have the option of executing a long-running
```

Comment:
Perhaps an example for how the krew plugin could call it? I can imagine as simple as this:
```
kubectl rook-ceph multus test ...
```

Reply:
I think that's something to decide for the plugin, but I would recommend just following the tool's CLI closely: `kubectl rook-ceph multus validation [run|cleaup]` is clear, and it allows for expanding the set of multus and multus validation commands in the future.